### PR TITLE
Lambda weather function

### PR DESCRIPTION
**Pull request by arno-hartmann**
Initiated Lamda and Terraform.
Open to Suggestions and Comments!

### CODE REVIEW

File: lambda_weather/build_function.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+#Todocp Python-Files to lambda_weather/
+#pip3 install --target ./package decimal
```
Comment (fabianschmauder) [inline lambda_weather/build_function.sh]:
```suggestion
```

File: lambda_weather/build_function.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+
+pip3 install --target ./package requests
+#pip3 install --target ./package datetime
```
Comment (fabianschmauder) [inline lambda_weather/build_function.sh]:
```suggestion
```

File: lambda_weather/src/get_weather.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import formatting_utils
+import get_weather_utils
+
+def get_weather_from_api():
+    today = date.today()
+
+    cities=[]
+    cities = get_weather_utils.get_all_destinations_from_dynamodb()
+
+    #counter = 0
```
Comment (fabianschmauder) [inline lambda_weather/src/get_weather.py]:
```suggestion
```

File: lambda_weather/src/get_weather.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    for item in cities:
+        for x in range(1,4):
+            day = today + timedelta( days = x)
+            day_str = formatting_utils.maketimestring(day)
+        #    api_url = get_weather_utils.determine_url_for_api(cities[counter]['city_id'], day_str)
+        #    get_weather_utils.write_weather_dynamoDB(api_url, cities[counter]['city'], x*100)
```
Comment (fabianschmauder) [inline lambda_weather/src/get_weather.py]:
```suggestion
```

File: lambda_weather/src/get_weather.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        
+            api_url = get_weather_utils.determine_url_for_api(item['city_id'], day_str)
+
+            get_weather_utils.write_weather_dynamoDB(api_url, item['city'], x*100)
+    #   counter+=1
```
Comment (fabianschmauder) [inline lambda_weather/src/get_weather.py]:
```suggestion
```

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+resource "aws_dynamodb_table" "weather" {
+  name           = "weather"
+  billing_mode   = "PROVISIONED"
+  read_capacity  = 20
+  write_capacity = 20
+  hash_key       = "city"
+  range_key      = "date"
+
+  #stream_enabled = true
+  #stream_view_type = "NEW_IMAGE"
```
Comment (fabianschmauder) [inline main.tf]:
```suggestion
```

File: main.tf [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+provider "aws" {
+  profile = "default"
+  region  = "us-west-2"
+}
+
+
+#HIER ANPASSEN?
+data "aws_caller_identity" "current" {}
+
+#HIER ANPASSEN?
```
Comment (fabianschmauder) [inline main.tf]: ?
```suggestion
data "aws_caller_identity" "current" {}

```

File: lambda_weather/src/get_weather.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    for item in cities:
+        for x in range(1,4):
+            day = today + timedelta( days = x)
+            day_str = formatting_utils.maketimestring(day)
+        #    api_url = get_weather_utils.determine_url_for_api(cities[counter]['city_id'], day_str)
+        #    get_weather_utils.write_weather_dynamoDB(api_url, cities[counter]['city'], x*100)
+        
+            api_url = get_weather_utils.determine_url_for_api(item['city_id'], day_str)
+
+            get_weather_utils.write_weather_dynamoDB(api_url, item['city'], x*100)
```
Comment (fabianschmauder) [inline lambda_weather/src/get_weather.py:19]: why x*100 ?

**Review by fabianschmauder — COMMENTED** [review]
a lot of commented lines

File: lambda_weather/src/get_weather.py (hunk 8 above)
Comment (arno-hartmann) [reply lambda_weather/src/get_weather.py:19]: It's an identifier for the different reports: 100+ tomorrow, 200+ in 2 days, 300+ in 3 days.

**Pull request merged**